tests/topotests/isis_srv6_te_topo1/test_isis_srv6_te_topo1.py

```diff
@@ -453,11 +456,208 @@ def test_srv6_te_policy_removed():
         !"
     )
 
-    for rname in ["rt1"]:
+    # for rname in ["rt1"]:
+    #    router_compare_json_output(
+    #        rname,
+    #        "show ipv6 route static json",
+    #        "step4/show_ipv6_route.ref",
+    #    )
+
+
+#
+# Step 5
+#
+# Test SRv6 End.B6.Encaps
+#
+
+
+def test_srv6_end_b6_encaps():
+    logger.info("Test (step 5): verify SRv6 END.B6.Encaps")
+    tgen = get_topogen()
+
+    # Skip if previous fatal error condition is raised
+    if tgen.routers_have_failure():
+        pytest.skip(tgen.errors)
+
+    tgen.gears["rt1"].vtysh_cmd(
+        "configure \n \
+         ipv6 route fc00:0:6b::/48 fc00:0:6:: color 1 \n \
+		 segment-routing \n \
+		 traffic-eng \n \
+ 		 segment-list srv6 \n \
+		  index 1 ipv6-address fc00:0:2:: \n \
+		  index 2 ipv6-address fc00:0:6:: \n \
+		  exit \n \
+		  policy color 1 endpoint fc00:0:6:: \n \
+		   candidate-path preference 1 name srv6 explicit segment-list srv6 \n \
+		 exit \n \
+		exit \n \
+		!"
+    )
+
+    tgen.gears["rt2"].vtysh_cmd(
+        "configure \n \
+		 segment-routing \n \
+		 traffic-eng \n \
+		  segment-list srv6-header \n \
+		   index 1 ipv6-address fc00:0:4:: \n \
+		   index 2 ipv6-address fc00:0:5:: \n \
+		  exit \n \
+		  policy color 1 endpoint fc00:0:6:: \n \
+		   srv6-binding-sid fc00:0:2:: \n \
+		   candidate-path preference 1 name srv6 explicit segment-list srv6-header \n \
+		 exit \n \
+		exit \n \
+		!"
+    )
+
+    tgen.gears["rt5"].vtysh_cmd(
+        "configure \n \
+          interface sr0 \n \
+           ipv6 address fc00:0:5::/48 \n \
+          exit"
+    )
+
+    tgen.gears["rt5"].run("sysctl -w net.ipv6.conf.all.seg6_enabled=1")
+    tgen.gears["rt5"].run("sysctl -w net.ipv6.conf.default.seg6_enabled=1")
+    tgen.gears["rt5"].run("sysctl -w net.ipv6.conf.eth-rt4.seg6_enabled=1")
+
+    subprocess.check_call(["apt", "install", "-y", "python3-venv"])
+    if not os.path.isdir("/tmp/topotests/isis_srv6_te_topo1.test_isis_srv6_te_topo1/rt2/scapy-latest"):
+        subprocess.check_call(
+            ["git", "clone", "https://github.com/secdev/scapy.git", "/tmp/topotests/isis_srv6_te_topo1.test_isis_srv6_te_topo1/rt2/scapy-latest"]
+        )
```

Comment:
I wonder if it is acceptable to proceed to upgrading scapy.
This said I understand that scapy is needed to catch dataplane traffic...

I let open the discussion by hoping that scapy upgrade is acceptable.
However, if not acceptable, why don't we just watch for the seg6local route by doing the standard 'show ipv6 route' checks ?

Reply:
thanks for explanation, this behavior is ok to me.